### PR TITLE
Destroy tracker on next tick, so STOP event has time to be sent

### PR DESCRIPTION
**Pull request by feross**
For: https://github.com/feross/bittorrent-tracker/issues/190

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```
Comment (CraigglesO) [inline index.js]: This doesn't have the nice 'instant gratification' setup as you see but what I did to get EVENT 3 to send was:
1) https://github.com/feross/bittorrent-tracker/pull/193
2) Completely remove the self.tracker.destroy(cb) and don't add the process.nextTick.
At line 122 you already submit a 'stopped' request and the UDP Tracker client recognizes this and sets up a quick cleanup after it's done. I'll submit my own PR in a minute.

NOTE: I would keep 'self.tracker = null' for JS garbage collection.

Comment (feross) [inline index.js]: It's not enough to just set `self.tracker = null` for garbage collection. There are open sockets and other objects that need to have `destroy()` called on them. So, unfortunately, this approach won't work.

Comment (CraigglesO) [inline index.js]: Right, but the UDP Tracker calls it's own socket.close() method in cleanup();

self.tracker = null is just to empty the pointer.

I experimented last night and it works. That's why I submitted a PR for the cleanup time update for bittorrent-tracker.

TL;DR : line 122 of this code you send a 'close()' method. This method actually has it's own cleanup implemented.
Trust me, the script runs and turns off. Do you want me to submit a test code that works with logs as I described?

Comment (CraigglesO) [inline index.js]: I just realized what you meant. This handles more then UDP.

Let me go back to the drawing board.

Comment (feross) [inline index.js]: The socket.close() method in cleanup() isn't enough to fully cleanup the udp tracker client. There is a setInterval and some other stuff that needs to cleaned up too, or else there will be memory leaks.

> I experimented last night and it works.

Yep, it works from the server's perspective (the STOP event is received), but not calling the destroy() method isn't gonna fly.

I have a large PR coming in a minute that fixes all the client implementations (not just udp) so they wait up to 1s for pending requests to be sent before force destroying everything.

Comment (CraigglesO) [inline index.js]: Great! Sorry I was not more help. 

Sometime later I would love to know your WS implementation. I am almost done with my TCP protocol implementation and I want my WS to be compatible with yours. I'm assuming you just send data like BEP09 metadata.. a bencoded dictionary followed by the data.

Comment (feross) [inline index.js]: No! You were a great help. I appreciate the attempts to help improve things.

**Comment by feross** [comment]
This PR is superseded by https://github.com/feross/bittorrent-tracker/pull/194
